docs/HowToGuides/FirstPullRequest.md

```diff
@@ -0,0 +1,168 @@
+programmers all over the globe.
+:sparkles::child::student::woman_technologist::technologist::man_technologist::sparkles:
+
+This document provides a high-level overview of different parts of the
+contribution process.
+
+## How do I pick something to work on?
+
+In case you don't have something specific you'd like to work on, such as
+implementing something for a Swift Evolution pitch, you could start off by
+working on a bug labeled `StarterBug` on [Swift JIRA][]. If the bug hasn't
+been assigned to someone, check the comments in case someone has already
+started working on it. If not, feel free to assign it to yourself and start
+working on it!
+
+[Swift JIRA]: https://bugs.swift.org
+
+## Getting Help
+
+Usually, Starter Bugs try to provide some instructions to help you get started.
+In case those are missing, please ask the bug reporter for more detailed steps
+and they will be happy to help.
+
+Once you start working on the bug, you will inevitably end up having a lot of
+questions. Don't be afraid to ask for help! The codebase is large and wrapping
+your head around it will take time. For example, you might have questions like:
+
+- Where can I find documentation on X?
+- I'm seeing a cryptic error E when trying to build the compiler. How do I fix
+  it or work around it?
+- I'm seeing very long build times even for incremental builds. How do I speed
+  up iteration time?
+- I'm not sure how to implement X. Any suggestions on where I should start?
+- What is the difference between types T1 and T2? They look very similar.
+- Should I split my new X into a separate file?
+- Should I create a new test file or update an existing test?
+- How should I test that I actually fixed this bug?
+- Test X is failing and I can't understand why. What might be going wrong here?
+- Test X is failing in CI but passing locally. Any tips for debugging?
+- I made some change but that seems to be not getting picked up. What should
+  I do to fix it?
+- I need to update the CMake but I'm not familiar with CMake. Could you give me
+  more guidance?
+- How do I do X in git?
+
+Some of these already have answers in our [FAQ](/docs/HowToGuides/FAQ.md).
+Maybe you have a question that's not on this list. That's fine.
+We're here to help. There are a couple of options to ask for help:
+
+- [Development category on the Swift forums](https://forums.swift.org/c/development):
+  Prefer using the forums for broad questions, such as those related to
+  building the toolchain, or understanding how something works at a high-level.
+  Since more people are likely to see and be able to answer your question, the
+  question is likely to get an answer sooner. Another benefit of asking in
+  public is that the answers you receive will be helpful to bystanders too.
+- Bug report/Pull request comments: Prefer asking in the bug report/pull request
+  when the question involves additional context specific to the
+  bug report/pull request.
```

Comment:
When I was new, I actually had more luck with Twitter than with the official channels. I don't think we want to suggest that specifically, but maybe we could say something like:

```markdown
- Other channels: Some Swift contributors are happy to answer questions or give help
  on social media. Not every contributor (or every social media account account with
  a contributor's name!) does this, though, so please check their profile and recent
  posts before approaching them outside of official channels, and please respect
  their wishes if they ask to move the discussion to a different forum.
```

Reply:
I would rather us not call this out. There's too much potential for it to be invasive on folks who don't want their social media and their work to collide.